api/v1alpha4/clusterclass_webhook.go

```diff
@@ -194,22 +200,95 @@ func (in *ClusterClass) validate(old *ClusterClass) error {
 		)
 	}
 
-	for _, class := range in.Spec.Workers.MachineDeployments {
-		for _, oldClass := range old.Spec.Workers.MachineDeployments {
-			if class.Class == oldClass.Class && !reflect.DeepEqual(class, oldClass) {
-				allErrs = append(allErrs,
-					field.Invalid(
-						field.NewPath("spec", "workers", "machineDeployments"),
-						class,
-						"cannot be changed.",
-					),
-				)
-			}
+	return allErrs
+}
+
+func (r LocalObjectTemplate) validate(namespace string, pathPrefix *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// check if a name is provided
+	if r.Ref.Name == "" {
+		allErrs = append(allErrs,
+			field.Invalid(
+				pathPrefix.Child("ref", "name"),
+				r.Ref.Name,
+				"cannot be empty",
+			),
+		)
+	}
+
+	// validate if namespace matches the provided namespace
+	if namespace != "" && r.Ref.Namespace != namespace {
```

Comment:
No strong opinion on that, but namespace is always != "", so we could drop the namespace != "".

Reply:
Going to keep it since `namespace` here is an argument to the function. Good to have a safe check. :)